cmd/shipperctl/cmd/backup/prepare.go

```diff
@@ -0,0 +1,86 @@
+package backup
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/bookingcom/shipper/cmd/shipperctl/configurator"
+	"github.com/bookingcom/shipper/cmd/shipperctl/release"
+)
+
+var (
+	prepareBackupCmd = &cobra.Command{
+		Use:   "prepare",
+		Short: "Get yaml of application and release objects prepared for backup",
+		Long: "Get yaml of application and release objects prepared for backup" +
+			"where every application will be bundled with it's releases",
+		RunE: runPrepareCommand,
+	}
+)
+
+func init() {
+	BackupCmd.AddCommand(prepareBackupCmd)
+}
+
+func runPrepareCommand(cmd *cobra.Command, args []string) error {
+	kubeClient, err := configurator.NewKubeClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	shipperClient, err := configurator.NewShipperClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	namespaceList, err := kubeClient.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	var errList []string
+
+	releasesPerApplications := []releasesPerApplication{}
+	for _, ns := range namespaceList.Items {
+		applicationList, err := shipperClient.ShipperV1alpha1().Applications(ns.Name).List(metav1.ListOptions{})
+		if err != nil {
+			errList = append(errList, err.Error())
+			continue
+		}
+
+		for _, app := range applicationList.Items {
+			releaseList, err := release.ReleasesForApplication(app.Name, app.Namespace, shipperClient)
+			if err != nil {
+				errList = append(errList, err.Error())
+				continue
+			}
+			releasesPerApplications = append(
+				releasesPerApplications,
+				releasesPerApplication{
+					Application: app,
+					Releases:    releaseList.Items,
+				},
+			)
+		}
+	}
+
+	data, err := printAndUnmarshalReleasesPerApplications(releasesPerApplications)
+	if err != nil {
+		errList = append(errList, err.Error())
+	} else {
+		err := ioutil.WriteFile(backupFile, data, 0644)
+		if err != nil {
+			errList = append(errList, err.Error())
+		}
+	}
+
+	if len(errList) > 0 {
+		return fmt.Errorf(strings.Join(errList, ", "))
```

Comment:
If there is no error at `printAndUnmarshalReleasesPerApplications`, but errors when retrieving some releases, the file will be created, but the command will crash with errors. Is that expected?

What about either:
- emit warnings when failing to retrieve releases of an application, and crashing if the `printAndUnmarshalReleasesPerApplications` is failing
or
- crashing at L57 to the first faill to retrieve releases of an application

Reply:
Solid point
How about if there were errors retrieving objects, warn user and ask to continue or stop? And if user chooses to continue, emit those warnings

Reply:
That works for me!